File: app/src/main/java/ru/svolf/architectre/lesson2/DataController.java
package ru.svolf.architectre.lesson2;

import androidx.lifecycle.MutableLiveData;

public class DataController {
    private static DataController instance;
    private final MutableLiveData<Integer> liveData = new MutableLiveData<>();

    public static DataController getInstance() {
        if (instance == null){
            instance = new DataController();
        }
        return instance;
    }

    public MutableLiveData<Integer> getData(){
        return liveData;
    }
}

File: app/src/main/java/ru/svolf/architectre/lesson2/DataController.kt
package ru.svolf.architectre.lesson2

import androidx.lifecycle.MutableLiveData

class DataController {
    val data = MutableLiveData<Int>()

    companion object {
        var instance: DataController? = null
            get() {
                if (field == null) {
                    field = DataController()
                }
                return field
            }
            private set
    }
}